### PR TITLE
EVG-6108 Renamed WhitelistService to WhitelistDataService and added some tests.

### PR DESCRIPTION
**Pull request by jimoleary**

### CODE REVIEW

**Comment by jimoleary** [comment]
trend chart support will follow on as the next commit.

File: public/static/app/common/WhitelistDataService.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      const outliers_collection = db
+        .db(STITCH_CONFIG.PERF.DB_PERF)
+        .collection(STITCH_CONFIG.PERF.COLL_WHITELISTED_OUTLIERS);
+
+      const promises = task_revisions.map(function(task_revision){
+        const query = _.pick(task_revision, 'revision', 'project', 'variant', 'task');
+        return outliers_collection.updateOne(query, task_revision, {upsert: true});
+      });
+      return $q.all(promises);
+    })
```
Comment (dbradf) [inline public/static/app/common/WhitelistDataService.js]: [nit] semicolon?

File: public/static/app/common/WhitelistDataService.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  // Removes whitelist entries for tasks
+  // :param task_revisions: list of task_revisions, containing 'revision', 'project', 'variant', 'task' values.
+  function removeWhitelist(task_revisions) {
+    const query = {$or: task_revisions.map((task_revision) => _.pick(task_revision, 'revision', 'project', 'variant', 'task'))};
+    return conn.query(function(db) {
+      return db
+        .db(STITCH_CONFIG.PERF.DB_PERF)
+        .collection(STITCH_CONFIG.PERF.COLL_WHITELISTED_OUTLIERS)
+        .deleteMany(query);
+    })
```
Comment (dbradf) [inline public/static/app/common/WhitelistDataService.js]: [nit] semicolon?

File: public/static/app/common/WhitelistDataService.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        $log.error('Cannot load outliers!', err);
+        return [];
+      });
+  };
+
+  return {
+    addWhitelist: addWhitelist,
+    removeWhitelist: removeWhitelist,
+    getWhitelistQ:getWhitelistQ,
+  }
```
Comment (dbradf) [inline public/static/app/common/WhitelistDataService.js]: [nit] semicolon?

File: public/static/app/common/WhitelistDataService.test.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        });
+
+        inject($injector => {
+          $q = $injector.get('$q');
+          deferred = $q.defer();
+          spyOn($q, 'all').and.returnValue(deferred.promise);
+
+          service = $injector.get('WhitelistDataService')
+        });
+
```
Comment (dbradf) [inline public/static/app/common/WhitelistDataService.test.js]: [nit] Can we remove some of these blank lines?

File: public/static/app/common/WhitelistDataService.test.js [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+      });
+      it('should process a single task_revisions', function () {
+        const task_revisions = setUpTaskRevisions();
+        const query  = createTaskRevisionQuery();
+
+        service.addWhitelist(task_revisions);
+        expect($db.db).toHaveBeenCalledWith(db_perf);
+        expect($db.collection).toHaveBeenCalledWith(coll_whitelist);
+        expect($db.updateOne).toHaveBeenCalledWith(query, task_revisions[0], {upsert: true});
+      });
```
Comment (dbradf) [inline public/static/app/common/WhitelistDataService.test.js]: [nit] In this file, it seems like sometimes we have a blank line between 'it' blocks and sometimes we don't. Can we pick one and do it everywhere? I'd vote for having a single blank line between 'it' blocks, but mostly care that we are consistent.

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js [hunk 6]
```diff
@@ -213,19 +243,17 @@ mciModule.controller('PerfBBRejectsCtrl', function (
...
       }));
+      api.selection.on.rowSelectionChangedBatch(null, rowSelectionChange);
 
-      // This is required when user selects all items
-      // (rowSelectionChanged doesn't work)
-      api.selection.on.rowSelectionChangedBatch(null, () => handleRowSelectionChange(api));
-
-      // Load initial set of data once `columns` are populated
-      api.core.on.rowsRendered(null, _.once(() => vm.reload()));
+      // api.core.on.filterChanged($scope, _.debounce(vm.reload, 200));
```
Comment (dbradf) [inline public/static/app/perf-bb/PerfBBRejectsCtrl.js]: [nit] Can we remove the commented out line?

File: public/static/app/common/WhitelistDataService.js (hunk 1 above)
Comment (jimoleary) [reply public/static/app/common/WhitelistDataService.js]: :+1:

File: public/static/app/common/WhitelistDataService.js (hunk 2 above)
Comment (jimoleary) [reply public/static/app/common/WhitelistDataService.js]: :+1:

File: public/static/app/common/WhitelistDataService.js (hunk 3 above)
Comment (jimoleary) [reply public/static/app/common/WhitelistDataService.js]: :+1:

File: public/static/app/common/WhitelistDataService.test.js (hunk 4 above)
Comment (jimoleary) [reply public/static/app/common/WhitelistDataService.test.js]: sure.

File: public/static/app/common/WhitelistDataService.test.js (hunk 5 above)
Comment (jimoleary) [reply public/static/app/common/WhitelistDataService.test.js]: Added blank lines between block.

But the real solution here is a linter.

File: public/static/app/perf-bb/PerfBBRejectsCtrl.js (hunk 6 above)
Comment (jimoleary) [reply public/static/app/perf-bb/PerfBBRejectsCtrl.js]: done

**Pull request merged**